File: trunk/quests/potimeb/Vallon_Zek.pl
my $x;
my $y;
my $z;
my $h;

sub EVENT_AGGRO {
       quest::say("Welcome to your final stand, mortals. You shall progress no farther!");
}

sub EVENT_SPAWN {
       quest::setnexthpevent(50);
       quest::settimer("vzek",1);
}

sub EVENT_HP {       
    if($hpevent == 50) {
        my $x = $npc->GetX();
        my $y = $npc->GetY();
        my $z = $npc->GetZ();
        my $h = $npc->GetHeading();
        quest::spawn2(223164,0,0,$x,$y - 25,$z,$h);
        quest::spawn2(223164,0,0,$x,$y + 25,$z,$h);
      
}
 }

sub EVENT_SIGNAL {
  if ($signal == 66) {
    quest::depop();
} 
 }

sub EVENT_DEATH {
        quest::signalwith(223157,12011,0);
        quest::say("You stand to gain no honor in my death. May the wrath of Rallos Zek strike you down with great vengeance.");
        quest::stoptimer("vzek");
}

sub EVENT_TIMER {
 my $x = $npc->GetX();
 my $y = $npc->GetY();
 if($timer eq "vzek") { 
 if($x < 260) {
    $npc->GMMove(405,75,358,192);
  }
 }
}

File: quests/potimeb/Vallon_Zek.lua
function event_death_complete(e)
	-- send a signal to the zone_status that I died
	eq.signal(223097,5);
end

function event_spawn(e)
	eq.set_next_hp_event(51);
end

function event_hp(e)
	local xloc = e.self:GetX();
	local yloc = e.self:GetY();
	local zloc = e.self:GetZ();
	local heading = e.self:GetHeading();
	if (e.hp_event == 51) then
		-- spawn copies
		eq.spawn2(223164,0,0,xloc,yloc - 25,zloc,heading);
		eq.spawn2(223164,0,0,xloc,yloc + 25,zloc,heading);
	end
end
